File: V.Vennela/Basics of Programming/Recursion and Backtracking/Introduction To Recursion/Factorial.java
import java.io.*;
import java.util.*;

public class Factorial{

    public static void main(String[] args) throws Exception {
        // write your code here
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        System.out.println(factorial(n));
    }

    public static int factorial(int n)
    {
        if(n==0 || n==1)
            return 1;
        
        return n*factorial(n-1);
    }

}

File: ayesha/Basic Data Structures/Recursion and Backtracking/Introduction to Recursion/factorial.cpp
#include <iostream>
using namespace std;

int factorial(int n)
{
    // write your code here
    if (n == 0 || n == 1)
    {
        return 1;
    }
    return n * factorial(n - 1);
}

int main()
{
    int n;
    cin >> n;
    cout << factorial(n);
}
